code/SerialPortStream.cs

```diff
@@ -699,9 +700,26 @@ private int InternalRead(byte[] buffer, int offset, int count)
             return bytes;
         }
 
-#if !NETSTANDARD15
-        private delegate int ReadDelegate(byte[] buffer, int offset, int count);
-
+#if NETSTANDARD15

```

Comment:
There's no reason this needs to be NETSTANDARD only, it should work just as well for .NET 4.5 and later. This would allow unit testing also for .NET 4.5. Honestly, the only reason I didn't have test cases for .NET Standard is because they were there for .NET FX and the differences were trivial.

Any chances to make this PR for both with NETSTANDARD15 || NET45, and then add your Unit Test cases at least for NET45, which should be the same/similar?

Reply:
> There's no reason this needs to be NETSTANDARD only, it should work just as well for .NET 4.5 and later

Correct. I will adapt it to use it for both, NETSTANDARD and .NET 4.5 or later.

>  and then add your Unit Test cases at least for NET45, which should be the same/similar

Yes, I'll add the test for .NET 4.5. My unit test has worked for .NET 4.5 before the change in this PR too, so the test wouldn't prove that it works for NETSTANDARD, but at least it shows that .NET 4.5 still works with these new changes.

Reply:
You're correct, you won't find the problem you found with .NET Standard by testing with .NET 4.5, but I hope there'll be at least some code coverage showing it works. Eventually, I'll get around to modernizing this a bit (upgrading to NUnit 3.x) and try to figure out how to get the unit tests working under dotnet core. That'll be later, and the unit tests will help none-the-less.